lib/mnesia/src/mnesia.erl

```diff
@@ -363,28 +404,60 @@ non_transaction(State, Fun, Args, ActivityKind, Mod)
 non_transaction(_State, Fun, Args, _ActivityKind, _Mod) ->
     {aborted, {badarg, Fun, Args}}.
 
+-spec async_dirty(Fun) -> Res | no_return() when
```

Comment:
It's a common misconception, but the type `foo() | no_return()` does not make much sense.  For starters, it's equivalent to just `foo()`.  Second, _all_ non-total functions in Erlang have an implicit no_return(), for example to handle all cases that they crash (or loop) for some reason.  I suggest you take out `| no_return()`.